### ISSUE TITLE
Crashing on Long press over link 

### ISSUE THREAD

**Issue by crazyhitty** [Saketme/Better-Link-Movement-Method#9]
Hi Saket, first of all thanks for creatng this awesome library, it makes much easier to manage links in my app.

But, I am facing this crash on long click.

```
FATAL EXCEPTION: main
Process: me.saket.betterlinkmovement.sample, PID: 4362
java.lang.StringIndexOutOfBoundsException: length=238; index=-1
    at java.lang.String.substring(String.java:1968)
    at android.text.SpannableStringInternal.<init>(SpannableStringInternal.java:33)
    at android.text.SpannableString.<init>(SpannableString.java:34)
    at android.text.SpannableString.subSequence(SpannableString.java:54)
    at me.saket.bettermovementmethod.BetterLinkMovementMethod$ClickableSpanWithText.ofSpan(BetterLinkMovementMethod.java:436)
    at me.saket.bettermovementmethod.BetterLinkMovementMethod.dispatchUrlLongClick(BetterLinkMovementMethod.java:395)
    at me.saket.bettermovementmethod.BetterLinkMovementMethod$1.onTimerReached(BetterLinkMovementMethod.java:232)
    at me.saket.bettermovementmethod.BetterLinkMovementMethod$LongPressTimer.run(BetterLinkMovementMethod.java:413)
    at android.os.Handler.handleCallback(Handler.java:789)
    at android.os.Handler.dispatchMessage(Handler.java:98)
    at android.os.Looper.loop(Looper.java:164)
    at android.app.ActivityThread.main(ActivityThread.java:6541)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
```

**Steps to reproduce:**
1. Click on a link.
2. Long click on the same link but just on text bounds boundary.

**Video:**
[![Crash video](https://img.youtube.com/vi/uvr9l5nDdaM/0.jpg)](https://www.youtube.com/watch?v=uvr9l5nDdaM)

**Comment by crazyhitty** [comment]
@Saketme This pull request closes this issue: https://github.com/Saketme/Better-Link-Movement-Method/pull/10

**Comment by Saketme** [comment]
Hey @crazyhitty, just letting you know that I'll have a look at this tomorrow. I'm traveling right now and the network here is really weak. Thanks for finding this bug though.

**Comment by bartwell** [comment]
Hi @Saketme,

This issue still exists. Could you please to fix it?

**Comment by Saketme** [comment]
@bartwell apologies for not being able to pay attention to this. I will fix this issue this week as soon as I get time.

**Comment by Saketme** [comment]
Fixed in commit [068976d](https://github.com/Saketme/Better-Link-Movement-Method/commit/068976dad49bb9ec1b67eeb71f2a55b508ac7ab6). Will release a new version soon. In the meanwhile, you can get a build off HEAD using [jitpack.io](jitpack.io).

**Comment by Saketme** [comment]
Fixed in 2.1.0: https://github.com/Saketme/Better-Link-Movement-Method/releases/tag/v2.1.0
